### PR TITLE
Fixed ability to skip interface implementations and union cases in query

### LINKED ISSUE #455: GraphQL Interface not implemented error when returning a list of interface types
This script demonstrates the problem:

```fsharp
#r "nuget: FSharp.Data.GraphQL.Server, 1.0.7"

type IPet =
  abstract member Name : string

type Dog =
  {
    Name : string
    Bark : int
  }
  interface IPet with
    member this.Name = this.Name

type Cat =
  {
    Name : string
    Agility : int
  }
  interface IPet with
    member this.Name = this.Name

open FSharp.Data.GraphQL
open FSharp.Data.GraphQL.Types

#nowarn "40"

let rec petType =
  Define.Interface<IPet>(
    name = "Pet",
    fields =
      [
        Define.Field("name", String, fun _ x -> x.Name)
      ],
    resolveType =
      fun x ->
        match x with
        | :? Dog -> dogType :> ObjectDef
        | :? Cat -> catType :> ObjectDef
        | _ -> failwith $"Invalid IPet object %s{x.GetType().FullName}"
  )

and dogType =
  Define.Object<Dog>(
    name = "Dog",
    fields =
      [
        Define.Field("name", String, fun _ x -> x.Name)
        Define.Field("bark", Int, fun _ x -> x.Bark)
      ],
    interfaces = [ petType ]
  )

and catType =
  Define.Object<Cat>(
    name = "Cat",
    fields =
      [
        Define.Field("name", String, fun _ x -> x.Name)
        Define.Field("agility", Int, fun _ x -> x.Agility)
      ],
    interfaces = [ petType ]
  )

let queryType =
  Define.Object<unit>(
    name = "Query",
    fields =
      [
        Define.Field(
          "pets",
          ListOf petType,
          fun _ () ->
            [
              { Name = "Fido"; Bark = 10 } :> IPet
              { Name = "Felix"; Agility = 8 } :> IPet
            ]
        )
      ]
  )

let config : SchemaConfig =
  {
    SchemaConfig.Default
      with
        Types =
          [
            dogType
            catType
          ]
  }

let schema = Schema(queryType, config = config)
let executor = Executor(schema)

let query =
  """
  {
    pets {
      ... on Dog {
        name
        bark
      }
    }
  }
  """

let result =
  executor.AsyncExecute(query)
  |> Async.RunSynchronously

printfn "%A" result
```

```
{ Content =
   Direct
     (seq
        [[documentId, -2037844746]; [data, ];
         [errors, [{ message: "GraphQL Interface 'Pet' is not implemented by the type 'Cat'",
		path: ["pets", 1, ] }]]],
      [("GraphQL Interface 'Pet' is not implemented by the type 'Cat'",
        ["pets"; 1])])
  Metadata = map [] }
```

Curiously, these queries work as expected:

```gql
{
  pets {
    name
    ... on Dog {
      bark
    }
  }
}
```

```gql
{
  pets {
    ... on Cat {
      name
      agility
    }
    ... on Dog {
      name
      bark
    }
  }
}
```

### PR DESCRIPTION
**Pull request by xperiandri**
Fixes #455

### CODE REVIEW

**Comment by xperiandri** [comment]
@njlr I fixed #455 
Please review

File: tests/FSharp.Data.GraphQL.Tests/AbstractionTests.fs [hunk 1]
```diff
@@ -111,6 +116,79 @@ let ``Execute handles execution of abstract types: isTypeOf is used to resolve r
...
+        ... on Dog {
+          name
+          woofs
+        }
+      }
+    }"""
+
+    let result = sync <| schemaWithInterface.Value.AsyncExecute (parse query)
+    ensureRequestError result <| fun [ petsError ] ->
+        petsError |> ensureExecutionError "Non-Null field pets resolved as a null!" [ "pets"; 1 ]
```
Comment (njlr) [inline tests/FSharp.Data.GraphQL.Tests/AbstractionTests.fs]: Is it correct that this produces an error? I would expect the result to be a list containing only this element: 

```json
{ 
  "name": "Odie",
  "woofs": true,
}
```

Maybe I have misunderstood the GraphQL spec, however?

Comment (xperiandri) [reply tests/FSharp.Data.GraphQL.Tests/AbstractionTests.fs]: I asked Copilot and it responded to me with nulls for not declared fragments.
Now I opened https://graphql.com/learn/interfaces-and-unions/#fragments and put this query
```graphql
query GetMostPopularStallProduce {
  mostPopularStall {
    availableProduce {
      ...  on Fruit {
        hasEdibleSeeds
      }
    }
  }
}
```
and it returns empty objects
```JSON
{
  "data": {
    "mostPopularStall": {
      "availableProduce": [
        {
          "hasEdibleSeeds": false
        },
        {
          "hasEdibleSeeds": true
        },
        {
          "hasEdibleSeeds": false
        },
        {},
        {}
      ]
    }
  }
}
```

Comment (xperiandri) [reply tests/FSharp.Data.GraphQL.Tests/AbstractionTests.fs]: which logically looks as the most appropriate option

Comment (xperiandri) [reply tests/FSharp.Data.GraphQL.Tests/AbstractionTests.fs]: @njlr do you have a link to the spec place where it is described?

Comment (xperiandri) [reply tests/FSharp.Data.GraphQL.Tests/AbstractionTests.fs]: @njlr have you found any?

Comment (njlr) [reply tests/FSharp.Data.GraphQL.Tests/AbstractionTests.fs]: I was unable to find anything in the spec! Perhaps we should open an issue on the spec repo?

Comment (xperiandri) [reply tests/FSharp.Data.GraphQL.Tests/AbstractionTests.fs]: Opened https://github.com/graphql/graphql-spec/issues/1080

Comment (xperiandri) [reply tests/FSharp.Data.GraphQL.Tests/AbstractionTests.fs]: @njlr I've fixed the logic accoding to the answer by the link above

Comment (xperiandri) [reply tests/FSharp.Data.GraphQL.Tests/AbstractionTests.fs]: If you agree, please, approve

File: src/FSharp.Data.GraphQL.Server/Execution.fs [hunk 2]
```diff
@@ -510,12 +515,12 @@ and private executeResolvers (ctx : ResolveFieldContext) (path : FieldPath) (par
     match info.Kind, returnDef with
     | ResolveDeferred innerInfo, _ when innerInfo.IsNullable -> // We can only defer nullable fields
         deferred
-        |> resolveWith { ctx with ExecutionInfo = { innerInfo with IsNullable = false } }
+        |> resolveWith { ctx with ExecutionInfo = innerInfo }
```
Comment (xperiandri) [inline src/FSharp.Data.GraphQL.Server/Execution.fs:518]: I don't know what was the idea here

**Comment by xperiandri** [comment]
@njlr and finally I'm going to release 2.0.0-beta1 after this PR 🙂

**Review by njlr — APPROVED** [review]

**Pull request merged**